### ISSUE TITLE
HTML::anchor

### ISSUE THREAD

**Issue by helpse** [kohana/core#503]
Hello, there is an issue with HTML::anchor

It doesnt allow you to use "?key=value" for href. The function always prepends a leading slash. It's because it uses internally URL::site. May be its there that it shold look for a leading '?' and do not prepend the slash.

**Comment by helpse** [comment]
This would be a better solution: (HTML::anchor, line 129)

```
elseif ($uri[0] !== '#' AND $uri[0] !== '?')
{
    // Make the URI absolute for non-id anchors
    $uri = URL::site($uri, $protocol, $index);
}
```

**Comment by enov** [comment]
On which version you are?

**Comment by helpse** [comment]
3.2, 3.3. Doesnt matter, its the same even in development branches.

**Comment by enov** [comment]
I guess you're making a point here. I'll mark this as a bug.

Please make a PR towards the development branches of 3.3 and 3.4. Also, add your tests, for both cases: the query part (?) and the fragment part (#):

https://github.com/kohana/core/blob/3.3/develop/tests/kohana/HTMLTest.php#L223

Thanks

**Comment by enov** [comment]
@helpse this is fixed in 3.3/develop.

Thanks for reporting!
